protocol/src/verifier/index.ts

```diff
@@ -281,6 +281,50 @@ class Handler implements Required<Protocol.Handler<boolean>> {
     return args.length === 2 && !!args[1];
   }
 
+  '|deinit|'(args: Args['|deinit|']) {
+    return args.length === 1;
+  }
+
+  '|selectorhtml|'(args: Args['|selectorhtml|']) {
+    return args.length === 3 && !!args[1] && !!args[2];
+  }
+
+  '|refresh|'(args: Args['|refresh|']) {
+    return args.length === 1;
+  }
+
+  '|tempnotify|'(args: Args['|tempnotify|']) {
+    return args.length >= 3 && !!args[1] && !!args[2] &&
+      (args.length === 3 ||
+        (args.length >= 4 && !!args[3] &&
+          (args.length === 4 || (args.length === 5 && !!args[4]))));
+  }
+
+  '|tempnotifyoff|'(args: Args['|tempnotifyoff|']) {
+    return args.length === 2 && !!args[1];
+  }
+
+  '|noinit|'(args: Args['|noinit|']) {
+    return args.length >= 3 && !!args[2] &&
+      (((args[1] === 'joinfailed' || args[1] === 'namerequired' || args[1] === 'nonexistent') &&
+          args.length === 3) ||
+        (args[1] === 'rename' && args.length === 4 && !!args[3]));
```

Comment:
note that now that we're using stricter types we actually should be able to use the existing helpers like `verifyRoomID` here (or `verifyID` below) etc.